pyglossary/plugins/ebook_kobo.py

```diff
@@ -162,14 +169,44 @@ def writeGroup(lastPrefix):
 				gzipFile.write(htmlContents.encode("utf-8"))
 			htmlContents = htmlHeader
 
-		lastPrefix = ""
+		data = []
+
 		while True:
 			entry = yield
 			if entry is None:
 				break
 			if entry.isData():
 				dataEntryCount += 1
 				continue
+			l_word = entry.l_word
+			if len(l_word) == 1:
+				data.append(entry.getRaw(glos))
+				continue
+			wordsByPrefix = OrderedDict()
+			for word in l_word:
+				prefix = self.get_prefix(word)
+				if prefix in wordsByPrefix:
+					wordsByPrefix[prefix].append(word)
+				else:
+					wordsByPrefix[prefix] = [word]
+			if len(wordsByPrefix) == 1:
+				data.append(entry.getRaw(glos))
+				continue
+			defi = entry.defi
+			for prefix, p_words in wordsByPrefix.items():
+				for word in l_word:
+					if word not in p_words:
+						p_words.append(word)
```

Comment:
This purpose of this is unclear to me. Do you want to add all keywords to each of the duplicated entries? The ones matching the prefix should be sufficient and are already added above, so this code code be skipped.

Reply:
Removed it.